policy/scan/local_scanner.go

```diff
@@ -395,6 +395,13 @@ func (s *LocalScanner) distributeJob(job *Job, ctx context.Context, upstream *up
 		}
 	}
 
+	// if there is exactly one asset, assure that the --asset-name is used
+	// TOOD: make it so that the --asset-name is set for the root asset only even if multiple assets are there
+	if len(assets) == 1 {
```

Comment:
we should be doing this earlier, this is post SynchronizeAssets

there's a check for `len(assets == 0)` somewhere up, just before the sync call.  You can also use `assetList` as that loads up the job inventory:
```
if len(assets) == 1 && assetsList[0].Name != assets[0].Name) {
 ...
 }
 ```
also do please write a comment that this is a temp fix